src/xrpld/app/tx/detail/DeleteAccount.cpp

```diff
@@ -202,19 +220,23 @@ DeleteAccount::preclaim(PreclaimContext const& ctx)
     if ((*sleDst)[sfFlags] & lsfRequireDestTag && !ctx.tx[~sfDestinationTag])
         return tecDST_TAG_NEEDED;
 
+    auto sleAccount = ctx.view.read(keylet::account(account));
+    assert(sleAccount);
+    if (!sleAccount)
+        return terNO_ACCOUNT;
```

Comment:
Why an assert _and_ returning the error?

Also, shouldn't this return `tefINTERNAL`?

Reply:
I didn't modify the code, just move it before credentials check.

Reply:
reverted